docs/source/utilities.rst

```diff
@@ -0,0 +1,72 @@
+Additionally, we would recommend adding ``source /opt/ros/foxy/setup.zsh`` to
+your ``.zshrc`` if using zsh or to your ``.bashrc``. So you don't have to
+type that out each time you open a new terminal. When in the future we
+transition to a different ROS version, you can update that line to the path
+of the new version.
+
+Launch
+--------------------------------------------------
+Our launch files are pretty similar to the ones introduced in ROS wiki
+tutorials, but we also use flags on launch to easily switch between certain
+nodes (for example we can switch between sim radio and network radio by
+setting a particular to the proper boolean value).
+Alias for common launch configurations can be found in the makefile.
+
+Build System
+--------------------------------------------------
+We use cmake and ninja. We prefer using clang as the C++ compiler compared to
+gcc, but either will work.
+Most of the details on the high-level construction of out build system can
+been ascertained by reading the root ``CMakeLists.txt``.
+If you are adding a new C++ file, it is best to just follow the existing
+format by reading through ``CMakeLists.txt`` in the relevant directories.
+
+Debugging C++ Code
+--------------------------------------------------
+Setting up a debugger for our C++ side code is actually quite simple!
+In particular, we have had success using `LLDB <https://lldb.llvm.org/>`_,
+made by the same group that develops the clang compiler.
+As of summer 2022, the ubuntu-setup script installs clang-10 using a script
+which also includes lldb-10
+(so you should not need to install anything new for this).
+All you need to do is start soccer or sim in one terminal tab, and in another
+tab, run ``sudo lldb-10 -n name_of_particular_proc``.
+
+How do you find this particular name?
+Well it depends on what file/node you wish to debug.
+As of the time of writing this article, we do not use any ros2 node
+composition, so each node is its own process.
+Looking in the soccer launch file a node's "executable_name" corresponds to
+the process name to place after ``-n``.
+Another method to find the process names of nodes is to run ``top`` in a new
+terminal tab and look in there.
+
+And how do you use lldb?
+Google what you want to do and follow the top result or follow the lldb
+tutorial on their website.
+CS2110 and CS2200 will introduce you to gdb which is another debugger for
+C/C++ and if you already know that,
+the commands are basically the same (the syntax is different in many places
+though).
+
+There is also another method which involves commenting out the node you want
+to run in the launch files and running it separately in a new terminal tab
+with a debugging prefix. This method is slower and not as consistent, so I
+won't explain it anymore here. Though check out ``util/debug-cpp.sh`` if you
+are curious or if the first method I explained above stops working for some
+reason.
+
+Continuous Integration
```

Comment:
I talked about this in the contributing page, but we can keep it here too. Or delete it? Up to you

Reply:
I guess we could also leave it here since it's a "utility."